[uv pip install "$WHEEL_FILENAME"] | Sanity tests nightly debug run with watcher | blackhole nightly debug run (Ubuntu 24.04 with watcher) / ttnn-unit-tests / ttnn fused group 1 [bh_p150b_civ2]

## CI Failure Report

<!-- AUTO-TRIAGE-SUMMARY-START -->
**Summary:** [CI] ttnn wheel install fails: built cp310 wheel incompatible with cp312 venv in blackhole nightly debug watcher run
<!-- AUTO-TRIAGE-SUMMARY-END -->
**Workflow:** Sanity tests nightly debug run with watcher
**Job:** blackhole nightly debug run (Ubuntu 24.04 with watcher) / ttnn-unit-tests / ttnn fused group 1 [bh_p150b_civ2]

### Regression timeline
- Last passing commit: N/A
- First failing commit: `5beed318d0f0` on 2026-07-31T01:31:31Z — https://github.com/tenstorrent/tt-metal/actions/runs/30596561396/job/91083380742
- Most recent failing commit: `db8bda8f1491` on 2026-08-01T01:31:05Z — https://github.com/tenstorrent/tt-metal/actions/runs/30678081493/job/91491115496

### Failing job URLs (last 2 runs)
- https://github.com/tenstorrent/tt-metal/actions/runs/30678081493/job/91491115496
- https://github.com/tenstorrent/tt-metal/actions/runs/30596561396/job/91083380742

### Failing test path(s)
- No tests ran; failure occurs during environment setup, in the step that installs the built ttnn wheel via `uv pip install "$WHEEL_FILENAME"` (before `ttnn fused group 1` tests execute).

### Error excerpt (terminal failure from logs)
```
📦 Installing ttnn-0.65.1rc17.dev7582+gdb8bda8f14-cp310-cp310-manylinux_2_35_x86_64.whl
Using Python 3.12.12 environment at: /opt/venv
  × No solution found when resolving dependencies:
  ╰─▶ Because only ttnn==0.65.1rc17.dev7582+gdb8bda8f14 is available and
      ttnn==0.65.1rc17.dev7582+gdb8bda8f14 has no wheels with a matching
      Python version tag (e.g., `cp312`), we can conclude that all versions of
      ttnn cannot be used.
      And because you require ttnn, we can conclude that your requirements
      are unsatisfiable.
##[error]Process completed with exit code 1.
```
(Run 2 shows the identical error for wheel `ttnn-0.65.1rc17.dev7536+g5beed318d0-cp310-cp310-manylinux_2_35_x86_64.whl` against the same `Python 3.12.12` venv.)

### Reproduction steps
1. Build the ttnn wheel artifact (`wheel-artifact-name: ttnn-dist-inplace-RelWithDebInfo-profiler-<sha>-<run_id>`) as produced by the build job for this workflow.
2. In the `ubuntu-24.04-ci-test-amd64` docker image, ensure a Python 3.12 (`/opt/venv`) environment is active.
3. Run `WHEEL_FILENAME=$(ls -1 *.whl); uv pip install "$WHEEL_FILENAME"`.
4. Observe `uv` fails to resolve because the wheel filename is tagged `cp310` (built for Python 3.10) while the target venv is Python 3.12 (`cp312`), producing "No solution found when resolving dependencies".

### Notes
- Both failures occur before any ttnn unit test executes — this is purely a Python ABI tag mismatch between the produced wheel (`cp310`) and the test venv (`cp312`), not a functional test regression.
- Determinism check:
  - Run 1 (job 91491115496, commit db8bda8f1491): terminal line `× No solution found when resolving dependencies: ... has no wheels with a matching Python version tag (e.g., \`cp312\`) ... ##[error]Process completed with exit code 1.`
  - Run 2 (job 91083380742, commit 5beed318d0f0): terminal line `× No solution found when resolving dependencies: ... has no wheels with a matching Python version tag (e.g., \`cp312\`) ... ##[error]Process completed with exit code 1.`
  - Both runs reference different commits/run URLs, satisfying the distinctness requirement for regression timeline boundaries.

<!-- AUTO-TRIAGE-METADATA-START -->
`Auto-triage-workflow: Sanity tests nightly debug run with watcher`
`Auto-triage-job-name: blackhole nightly debug run (Ubuntu 24.04 with watcher) / ttnn-unit-tests / ttnn fused group 1 [bh_p150b_civ2]`
<!-- AUTO-TRIAGE-METADATA-END -->

<!-- AUTO-TRIAGE-INFO-START -->
## About this issue (automated CI auto-triage)
This issue was opened and assigned by an automated AI system. It posts through a bot token, so notifications may look like they come from a person, but they are automated.

The owner is selected from recent commit history and job-ownership data. It reflects who is *likely able to fix or route* this, not necessarily who caused it, so it can be imperfect.

**If this was assigned to the wrong person, you can:**
- Comment `REASSIGN OWNER: <why this is outside your area / which team or test-area it belongs to>` and the system will automatically re-route it to a better-suited owner.
- Unassign yourself (this stops notifications to you).
- Close the issue if it is not a real or relevant failure.

The assigned owner is reminded about every 3 days until the issue is resolved. Full guide: https://tenstorrent.atlassian.net/wiki/spaces/MI6/pages/2545221692/Auto-Triage+Info
<!-- AUTO-TRIAGE-INFO-END -->

--- a/.github/workflows/blackhole-sanity-tests.yaml
+++ b/.github/workflows/blackhole-sanity-tests.yaml
@@ -150,7 +150,9 @@ jobs:
     with:
       harbor-prefix: ${{ needs.check-harbor.outputs.harbor-prefix }}
       build-type: ${{ inputs.build-type || 'Release' }}
-      build-inplace-wheel: true
+      # In-place wheel is cp310; only valid for 22.04 tests. 24.04 (cp312) needs the manylinux wheel.
+      build-inplace-wheel: ${{ (inputs.platform || 'Ubuntu 22.04') == 'Ubuntu 22.04' }}
+      build-wheel: ${{ (inputs.platform || 'Ubuntu 22.04') != 'Ubuntu 22.04' }}
       tracy: true
       platform: ${{ inputs.platform || 'Ubuntu 22.04' }}
       build-umd-tests: true

--- a/.github/workflows/sanity-tests-debug.yaml
+++ b/.github/workflows/sanity-tests-debug.yaml
@@ -98,7 +98,8 @@ jobs:
     with:
       harbor-prefix: ${{ needs.check-harbor.outputs.harbor-prefix }}
       build-type: RelWithDebInfo
-      build-inplace-wheel: true
+      # Manylinux wheel: in-place is cp310, but this leg tests on 24.04 (cp312 venv).
+      build-wheel: true
       tracy: true
       platform: "Ubuntu 24.04"
       skip-tt-train: false

